### ISSUE TITLE
testing image get wrong results

### ISSUE THREAD

**Issue by wujindou** [dalgu90/resnet-18-tensorflow#4]
Hi, I'm new to resnet18, I'm so appreciated about your code. I modified your file extractor_torch_t7.py (attached) to test the model on some pictures I chose but all classified wrong.  So I'm wondering about your testing results, pictures you used & performances? So could you give me some advice to improve my test performance? thx!
```
with tf.Graph().as_default():
    global_step = tf.Variable(0, trainable=False, name='global_step')
    images = [tf.placeholder(tf.float32, [2, 224, 224, 3])]
    labels = [tf.placeholder(tf.int32, [2])]

    # Build model
    print("Build ResNet-18 model")
    hp = resnet.HParams(batch_size=2,
                        num_gpus=1,
                        num_classes=1000,
                        weight_decay=0.001,
                        momentum=0.9,
                        finetune=False)
    network_train = resnet.ResNet(hp, images, labels, global_step, name="train")
    network_train.build_model()
    print('Number of Weights: %d' % network_train._weights)
    print('FLOPs: %d' % network_train._flops)

    # Build an initialization operation to run below.
    init = tf.global_variables_initializer()

    # Start running operations on the Graph.
    sess = tf.Session(config=tf.ConfigProto(
        gpu_options = tf.GPUOptions(per_process_gpu_memory_fraction=0.96),
        allow_soft_placement=True,
        log_device_placement=False))
    sess.run(init)

    # Set variables values
    print('Set variables to loaded weights')
    all_vars = tf.trainable_variables()
    for v in all_vars:
        print('\t' + v.op.name)
        assign_op = v.assign(model_weights[v.op.name])
        sess.run(assign_op)
    image_path='/images/'
    images =[] 
    for imgname in os.listdir(image_path):
    	img = cv2.imread(image_path+"/"+imgname)
    	img = cv2.resize(img, (224,224), interpolation=cv2.INTER_CUBIC)
    	img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
    	img = img.astype(np.uint8)
    pred= sess.run(network_val.preds)

    # Save as checkpoint
    print('Save as checkpoint: %s' % INIT_CHECKPOINT_DIR)
    #sys.exit(1)
    if not os.path.exists(INIT_CHECKPOINT_DIR):
        os.mkdir(INIT_CHECKPOINT_DIR)
    saver = tf.train.Saver(tf.global_variables())
    saver.save(sess, os.path.join(INIT_CHECKPOINT_DIR, 'model.ckpt'))

print('Done!')

```

**Comment by dalgu90** [comment]
Hi @wujindou, actually there are an issue regarding the code.

First, there are some wrong implementation of `extract_torch_t7.py` and `imagenet_input.py`(To give some details, the weights are not fully extracted in the first and there is some issue in image preprocessing in the second). So you should use the latest version of the code.
Actually, the preprocessing is not correctly done in your code snippet. You should 1. make the value scale is inbetween 0~1 (not 0~255), and 2. normalize per color channel(check `preprocess_image()` in `imagenet_input.py`).

I hope this help you.

**Comment by dalgu90** [comment]
And don't forget to feed images(and labels; If you don't need to compute accuracy internally, just feed zero values) when run `sess.run()`.

**Comment by yxk9810** [comment]
@dalgu90 ,Thanks for your help. The code works well now. I also want to know that , can i use this model  as a pretrained model and train on my own dataset ?

**Comment by dalgu90** [comment]
I didn't check that the code achieve the full performance(appeared in [here](https://github.com/facebook/fb.resnet.torch/blob/master/README.md#single-crop-224x224-validation-error-rate)) after finetuning. If you do, you can freely use it.
